### DISCUSSION TITLE
Radar Antenna Patterns and Electromagnetics propagation modelling

### DISCUSSION THREAD

**Discussion by LyceanEM** [dstl/Stone-Soup#500] [Ideas]
Hello

I have been looking through the code for the radar models, because I am interested in using my antenna array and propagation model to generate realistic antenna patterns, and I was hoping to be able to use those patterns within Stone Soup in order to investigate the effects of different beamforming architectures. However it looks like only the AESA sensor model has the required properties for beamwidth, antenna gain, etc. 

Is there any scope for the implementation of antenna pattern as a base property for radar sensor models? Ideally with a polarisation specification, but even without if this could be polled for the antenna gain in the appropriate direction of the targets it would be an improvement to the accuracy of the radar models.

**Comment by sdhiscocks** [comment]
Other than AESA sensor, most of our radar models are very simple at the moment, some having a simple max range property and fov, but not much else.

I certainly think it would be interesting to have the ability to add antenna patterns. I guess we'd also then have to add a attribute or agreed meta data value for things like target RCS (I think the AESA radar may already be using this) added to the ground truth type.

**Reply by LyceanEM** [reply]
Thank you for the reply. In that case I might have a look at writing a variant to RadarBearingRange or RadarRotatingBearingRange based upon the AESARadar object. Simply adding some form of more realistic antenna pattern such as a cosecant squared pattern, and using the AESARadar objects detection probability function, modified to call the attribute would improve matters. AESA is to complicated for now short of a dense matrix of the overall antenna pattern for all possible command angles.

**Comment by jjosborne-dstl** [comment]
The BeamShape objects are the current method for implementing beam shape. So far we've only a 2D Gaussian.
